tutorials-v5/miscellaneous/excitation-number-restricted-states-jc-chain.md

```diff
@@ -0,0 +1,264 @@
+
+$H_0 = \sum_{i=0}^{N} a_i^\dag a_i + s_i^\dag s_i$,
+
+the atom-cavity interactions:
+
+$H_{int,AC} = \sum_{i=0}^{N} = \frac{1}{2} (a_i^\dag s_i + s_i^\dag a_i)$,
+
+and the cavity-cavity interactions:
+
+$H_{int,CC} = \sum_{i=0}^{N-1} 0.9 \cdot (a_i^\dag a_{i+1} + a_{i+1}^\dag a_{i})$,
+
+where the interaction strength of $0.9$ was chosen arbitrarily.
+
+
+### Problem paramters
+
+```python
+N = 4  # number of systems
+M = 2  # number of cavity states
+dims = [M, 2] * N  # dimensions of JC spin chain
+excite = 1  # total number of excitations
+init_excite = 1  # initial number of excitations
+```
+
+### Setup to Calculate Time Evolution
+
+```python
+def solve(d, psi0):
+    # annihilation operators for cavity modes
+    a = d[::2]
+    # atomic annihilation operators
+    sm = d[1::2]
+
+    # when constructing the Hamiltonian notice the ordering of annihilation and creation operators (see introduction)
+    H0 = sum([aa.dag() * aa for aa in a]) + sum([s.dag() * s for s in sm])
+
+    # atom-cavity couplings
+    Hint_ac = 0
+    for n in range(N):
+        Hint_ac += 0.5 * (a[n].dag() * sm[n] + sm[n].dag() * a[n])
+
+    # cavity-cavity couplings
+    Hint_cc = 0
+    for n in range(N - 1):
+        Hint_cc += 0.9 * (a[n].dag() * a[n + 1] + a[n + 1].dag() * a[n])
+
+    H = H0 + Hint_ac + Hint_cc
+
+    e_ops = [x.dag() * x for x in d]
+    c_ops = [0.01 * x for x in a]
+
+    times = np.linspace(0, 250, 1000)
+    L = liouvillian(H, c_ops)
+    result = mesolve(
+        H, psi0, times, c_ops, e_ops, options={"nsteps": 5000, "store_states": True}
+    )
+    return result, H, L
+```
+
+### Regular QuTiP States and Operators
+
+```python
+d = [
+    tensor(
+        [
+            destroy(dim1) if idx1 == idx2 else identity(dim1)
+            for idx1, dim1 in enumerate(dims)
+        ]
+    )
+    for idx2, _ in enumerate(dims)
+]
+psi0 = tensor(
+    [
+        basis(dim, init_excite) if idx == 1 else basis(dim, 0)
+        for idx, dim in enumerate(dims)
+    ]
+)
+```
+
+```python
+%time res1, H1, L1 = solve(d, psi0)
+```
+
+### Using ENR States and Operators
+
+```python
+d = enr_destroy(dims, excite)
+psi0 = enr_fock(dims, excite, [init_excite if n == 1 else 0 for n in range(2 * N)])
+```
+
+```python
+%time res2, H2, L2 = solve(d, psi0)
+```
+
+### Comparison of Expectation Values
+
+```python
+fig, axes = plot_expectation_values([res1, res2])
+fig.set_figwidth(10)
+fig.set_figheight(8)
+for idx, ax in enumerate(axes):
+    if idx % 2:
+        ax.set_ylabel(f"Atom {idx//2}")
+    else:
+        ax.set_ylabel(f"Cavity {idx//2}")
+    ax.set_ylim(-0.1, 1.1)
+    ax.grid()
+fig.tight_layout()
+```
+
+### Calculation of Partial Trace
+
+The usage of ENR states makes many standard QuTiP features fail.
+```ptrace``` is one of those.
+Below we demonstrate how the partial trace for ENR states can be calculated and show the corresponding result together with the standrad QuTiP approach.
+
+```python
```

Comment:
did you mean to leave this in markup? i guess the tests will fail without explicitly defining it

Reply:
to be a bit clearer, when i convert this from markdown into a notebook this python part appears in the markdown cell above it, bit odd, am not sure why!

Reply:
I see, it was supposed to just be part of the text. I will make it italic then.